compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/packaging/converters/URIDryConverter.java

```diff
@@ -125,19 +125,22 @@ public URIDryConverter(URI base, Map<PackageID, Manifest> dependencyManifests, b
                         // get the version from the 'Location' header.
                         String location = conn.getHeaderField("Location");
                         String version = location.split("/")[location.split("/").length - 3];
-                        // update version
+                        // update version of moduleID
                         moduleID.version = new Name(version);
                         return Stream.empty();
                     } else if (statusCode == 400 && !loggedError) {
                         try (BufferedReader errorStream = new BufferedReader(
                                 new InputStreamReader(conn.getInputStream(), Charset.defaultCharset()))) {
                             String errorContent = errorStream.lines().collect(Collectors.joining("\n"));
-                            this.errStream.println("invalid request sent to remote registry: " + errorContent);
+                            this.errStream.println("error: could not connect to remote repository to find the latest " +
```

Comment:
Instead of manually printing this to std err, can't you create an instance of ErrorValue and then throw it? i.e., panic

Reply:
or is this not a Ballerina program?

Reply:
We dont want to break the module resolving because the central cannot be reached. Hence this should be fine IMO.